src/layouts/Page.tsx

```diff
@@ -191,6 +187,17 @@ export const Page = (props: Props) => {
     window.location.href = window.config.SIGNUP_URL ?? import.meta.env.VITE_SIGNUP_URL
   }
 
+  useEffect(() => {
+    const hasLifespanToken = user.childs.length > 0
+
+    if (hasLifespanToken && location.pathname === routes.getLifespanToken) {
+      history.push(routes.index)
```

Comment:
Cannot be done with using of React router?

Reply:
maybe in new version if React Router (with navigate()). I think it would be better to migrate to new version with new task